### PR TITLE
Error handling & notifier

### PR DESCRIPTION
**Pull request by dandehavilland**
PT: https://www.pivotaltracker.com/story/show/170894758

### CODE REVIEW

File: app/models/scheduler.rb [hunk 1]
```diff
@@ -1,7 +1,21 @@
...
   validates :facility_id, presence: true, numericality: { only_integer: true, greater_than: 0 }
   validates :batch_id, presence: true, numericality: { only_integer: true, greater_than: 0 }
   validates :integration_id, presence: true
   validates :run_on, presence: true
+  validates :attempts, numericality: { only_integer: true, less_than: MAX_ATTEMPTS }
+
+  def reschedule!
+    update!(
+      attempts: attempts + 1,
+      run_on: Time.now.utc + BACK_OFF.call(attempts)
```
Comment (dandehavilland) [inline app/models/scheduler.rb]: To be rescheduled for one hour later, up to `MAX_ATTEMPTS`.

File: app/jobs/scheduled_job.rb [hunk 2]
```diff
@@ -1,34 +1,65 @@
...
-            data: {
-              id: task.batch_id
-            }
-          },
-          facility: {
-            data: {
-              id: task.facility_id
-            }
+    rescue TooManyRetriesError
+      report_error
```
Comment (dandehavilland) [inline app/jobs/scheduled_job.rb]: Should also trigger a `warning` email to the CS team each time the task is rescheduled.

File: app/jobs/scheduled_job.rb [hunk 3]
```diff
@@ -1,34 +1,86 @@
...
-          batch: {
-            data: {
-              id: task.batch_id
-            }
-          },
-          facility: {
-            data: {
-              id: task.facility_id
-            }
+    rescue TooManyRetriesError => e
```
Comment (betacar) [inline app/jobs/scheduled_job.rb]: Is there a limit on retries? If not, let's have a hard limit of 4. Please. After that, CS should contact the customer and let them know this.

File: app/mailers/application_mailer.rb [hunk 4]
```diff
@@ -1,4 +1,4 @@
 class ApplicationMailer < ActionMailer::Base
-  default from: 'from@example.com'
+  default from: 'Artemis Support <support@artemisag.com>'
```
Comment (betacar) [inline app/mailers/application_mailer.rb]: Let's have this email address as an ENV var, please.

File: app/models/scheduler.rb [hunk 5]
```diff
@@ -1,7 +1,22 @@
 class Scheduler < ApplicationRecord
+  MAX_ATTEMPTS = 4
```
Comment (betacar) [inline app/models/scheduler.rb]: Nice!

File: db/schema.rb [hunk 6]
```diff
@@ -57,6 +57,7 @@
     t.integer "batch_id", null: false
     t.datetime "run_on", null: false
     t.datetime "received_at", default: -> { "CURRENT_TIMESTAMP" }, null: false
+    t.integer "attempts", default: 0
```
Comment (betacar) [inline db/schema.rb]: 💥

File: app/jobs/scheduled_job.rb (hunk 3 above)
Comment (betacar) [reply app/jobs/scheduled_job.rb]: nvm

**Pull request merged**